docs/opentelemetry-tracing.rst

```diff
@@ -9,32 +9,84 @@ To take advantage of these traces, we first need to install OpenTelemetry:
 .. code-block:: sh
 
     pip install opentelemetry-api opentelemetry-sdk opentelemetry-instrumentation
-
-    # [Optional] Installs the cloud monitoring exporter, however you can use any exporter of your choice
-    pip install opentelemetry-exporter-google-cloud
+    pip install opentelemetry-exporter-gcp-trace
 
 We also need to tell OpenTelemetry which exporter to use. To export Spanner traces to `Cloud Tracing <https://cloud.google.com/trace>`_, add the following lines to your application:
 
 .. code:: python
 
     from opentelemetry import trace
     from opentelemetry.sdk.trace import TracerProvider
-    from opentelemetry.trace.sampling import ProbabilitySampler
+    from opentelemetry.sdk.trace.sampling import TraceIdRatioBased
     from opentelemetry.exporter.cloud_trace import CloudTraceSpanExporter
-    # BatchExportSpanProcessor exports spans to Cloud Trace 
+    # BatchSpanProcessor exports spans to Cloud Trace
     # in a seperate thread to not block on the main thread
-    from opentelemetry.sdk.trace.export import BatchExportSpanProcessor
+    from opentelemetry.sdk.trace.export import BatchSpanProcessor
 
     # Create and export one trace every 1000 requests
-    sampler = ProbabilitySampler(1/1000)
+    sampler = TraceIdRatioBased(1/1000)
     # Use the default tracer provider
     trace.set_tracer_provider(TracerProvider(sampler=sampler))
     trace.get_tracer_provider().add_span_processor(
         # Initialize the cloud tracing exporter
-        BatchExportSpanProcessor(CloudTraceSpanExporter())
+        BatchSpanProcessor(CloudTraceSpanExporter())
+    )
+
+
+Alternatively you can pass in a tracer provider into the Cloud Spanner initialization
+
+.. code:: python
+
+    from opentelemetry.sdk.trace import TracerProvider
+    from opentelemetry.sdk.trace.sampling import TraceIdRatioBased
+    from opentelemetry.exporter.cloud_trace import CloudTraceSpanExporter
+
+    # Create and export one trace every 1000 requests
+    sampler = TraceIdRatioBased(1/1000)
+    tracerProvider = TracerProvider(sampler=sampler)
+    tracerProvider.add_span_processor(
+        # Initialize the cloud tracing exporter
+        BatchSpanProcessor(CloudTraceSpanExporter())
     )
 
+    o11y = dict(tracer_provider=tracerProvider)
+    # Pass the tracer provider while creating the Spanner client.
+    spanner_client = spanner.Client(observability_options=o11y)
+    instance = spanner_client.instance(instance_id)
+    database = instance.database(database_id)
+
+please note that the tracer being used is retrieved by invoking:
+
+.. code:: python
+
+   tracer = tracerProvider.get_tracer('cloud.google.com/python/spanner', SPANNER_LIB_VERSION)
```

Comment:
As per opentelemetry, this module name `cloud.google.com/python/spanner ` and version `SPANNER_LIB_VERSION ` are used to identify which library is the trace getting generated from. For ex, traces generated in python spanner client will be identified through these variables.

For customers if they want to add custom trace in their application, they will need to use their package name and application version that will help them understand that the custom trace is getting emited from their application.
There is no usecase for customer to retrieve the tracer with these python spanner package names. So you can remove this example here